src/Laravel/Mixins/FileMixin.php

```diff
@@ -20,21 +22,25 @@ public function saveToDisk()
         return function (string $path, string $disk = null, array $clientOpt = []): bool {
             /** @var File $this */
 
-            if (is_dir($path)) {
-                $path = rtrim($path, DIRECTORY_SEPARATOR).DIRECTORY_SEPARATOR;
+            $storage = Storage::disk($disk);
```

Comment:
Maybe move this to a trait or something, but the code is almost duplicated between the mixins as codeclimate reports